src/sentry/api/authentication.py

```diff
@@ -292,37 +341,62 @@ def accepts_auth(self, auth: list[bytes]) -> bool:
         return not token_str.startswith(SENTRY_ORG_AUTH_TOKEN_PREFIX)
 
     def authenticate_token(self, request: Request, token_str: str) -> tuple[Any, Any]:
-        token = SystemToken.from_request(request, token_str)
-        try:
-            token = (
-                token
-                or ApiToken.objects.filter(token=token_str)
-                .select_related("user", "application")
-                .get()
-            )
-        except ApiToken.DoesNotExist:
+        user: AnonymousUser | RpcUser | None = AnonymousUser()
+
+        token: SystemToken | ApiTokenReplica | ApiToken | None = SystemToken.from_request(
+            request, token_str
+        )
+
+        application_is_inactive = False
+
+        if not token:
+            if SiloMode.get_current_mode() == SiloMode.REGION:
+                atr = token = ApiTokenReplica.objects.filter(token=token_str).last()
+                if not atr:
+                    raise AuthenticationFailed("Invalid token")
+                user = user_service.get_user(user_id=atr.user_id)
+                application_is_inactive = not atr.application_is_active
+            else:
+                try:
+                    at = token = (
+                        ApiToken.objects.filter(token=token_str)
+                        .select_related("user", "application")
+                        .get()
+                    )
+                except ApiToken.DoesNotExist:
+                    raise AuthenticationFailed("Invalid token")
+                user = at.user
+                application_is_inactive = (
+                    at.application is not None and not at.application.is_active
+                )
```

Comment:
Same comment on this being possibly moved to the auth service impl

Reply:
Hmmm.... I guess, not?  My intention here is to keep the full implementation consolidated.  The "auth service" really has much to do with the RPC elements of the HC implementation.  My hope here is to ensure that the singular Token lookup logic is in one, clear place, and not divided.  For instance, it really isn't HC's job to say anything about ho tokens are used for auth, and that's the goal of this PR.  That said, I'm open to refactoring these conditionals into separate methods if that would look better.